### PR TITLE
Add creative paste container - Issue #211

### LINKED ISSUE #211: Creative Construction Paste container
I would like to offer the thought of a creative construction paste container which is always half full.

This would allow someone to go into creative mode, get only the paste container and then go back into survival to do builds where they don't want to farm the blocks themselves.  By having creative paste, they can paste as large a structure as they want, and not have to worry about anything but power/experience.  And when they break the building, they get paste back which then goes into the container, meaning they get essentially nothing back.

If you wanted to take things a step further, you could keep track of which blocks were "creative paste" and have them break and drop nothing at all, i.e. creative mode block breaking.

### PR DESCRIPTION
**Pull request by hnOsmium0001**
Fixes #211 - adds a creative paste container.

+ Slightly changed `setPasteAmount`/`getPasteAmount` logic that the static method delegates to the method `setPastes` and `getPastes`, which allows polymorphism.
+ Texture might need to be redone, I rubbed it together in 1 minute.

### CODE REVIEW

File: src/main/java/com/direwolf20/buildinggadgets/common/items/GenericPasteContainer.java [hunk 1]
```diff
@@ -23,16 +23,46 @@ public GenericPasteContainer() {
...
+     * Helper method.
+     * Delegates to {@link GenericPasteContainer#getPastes(ItemStack)}}.
+     */
+    public static int getPasteAmount(ItemStack stack) {
+        Item item = stack.getItem();
+        if (item instanceof GenericPasteContainer) {
+            return ((GenericPasteContainer) item).getPastes(stack);
+        } else {
+            BuildingGadgets.logger.warn("Potential abuse of GenericPasteContainer#getPasteAmount(ItemStack) where the given ItemStack does not contain a GenericPasteContainer.");
+            return 0;
```
Comment (Phylogeny) [inline src/main/java/com/direwolf20/buildinggadgets/common/items/GenericPasteContainer.java]: Lines 48-49 are unnecessarily nested in an else clause.

File: src/main/java/com/direwolf20/buildinggadgets/common/items/GenericPasteContainer.java [hunk 2]
```diff
@@ -23,16 +23,46 @@ public GenericPasteContainer() {
...
+        }
+    }
+
+
```
Comment (Phylogeny) [inline src/main/java/com/direwolf20/buildinggadgets/common/items/GenericPasteContainer.java]: Duplicate line break.

File: src/main/java/com/direwolf20/buildinggadgets/common/items/GenericPasteContainer.java [hunk 3]
```diff
@@ -43,6 +73,7 @@ public static int getPasteAmount(ItemStack stack) {
         return amount;
     }
 
+
```
Comment (Phylogeny) [inline src/main/java/com/direwolf20/buildinggadgets/common/items/GenericPasteContainer.java]: Duplicate line break.

File: src/main/java/com/direwolf20/buildinggadgets/common/items/GenericPasteContainer.java [hunk 4]
```diff
@@ -23,16 +23,46 @@ public GenericPasteContainer() {
...
+    public void setPastes(ItemStack stack, int amount) {
         NBTTagCompound tagCompound = stack.getTagCompound();
         if (tagCompound == null) {
             tagCompound = new NBTTagCompound();
         }
         tagCompound.setInteger("amount", amount);
         stack.setTagCompound(tagCompound);
+
+        stack.setTagCompound(tagCompound);
```
Comment (Phylogeny) [inline src/main/java/com/direwolf20/buildinggadgets/common/items/GenericPasteContainer.java]: Duplicate method call and unnecessary line break.

**Comment by Phylogeny** [comment]
A branch that's requested to merge as a PR must not be deleted, if the PR is to remain open.

**Comment by hnOsmium0001** [comment]
Yes I messed up when I was trying to merge all my local branches, and then it left me with no choice than opening a new PR. Or maybe it's just I don't know how to do it.
Sorry for that, again

**Pull request closed without merging**